src/platform/sdl2-platform.c

```diff
@@ -388,8 +388,9 @@ static void _gameLoop() {
 static boolean _pauseForMilliseconds(short ms) {
     SDL_UpdateWindowSurface(Win);
     SDL_Delay(ms);
-    return (pollBrogueEvent(&lastEvent, false) || lastEvent.eventType != EVENT_ERROR)
-        && lastEvent.eventType != MOUSE_ENTERED_CELL;
+    if (lastEvent.eventType == EVENT_ERROR || lastEvent.eventType == MOUSE_ENTERED_CELL)
+        pollBrogueEvent(&lastEvent, false);
```

Comment:
Ok, so the `lastEvent.eventType == MOUSE_ENTERED_CELL` is needed so we don't end up in a pause loop, returning false while a MOUSE_ENTERED_CELL event sits in lastEvent?

If you can amend and force-push adding braces I'll merge (sorry, yet to write a proper style guide)
```suggestion
    if (lastEvent.eventType == EVENT_ERROR || lastEvent.eventType == MOUSE_ENTERED_CELL) {
        pollBrogueEvent(&lastEvent, false);
    }
```

Reply:
Now that you mention it, we don't seem to need `MOUSE_ENTERED_CELL` at all in this function. It works better without it, in fact: buttons in the main menu react to mouseover. Any idea why it was there?

Reply:
This should work better now. I tested:

- [x] Main menu buttons react to mouseover
- [x] Mouse movements do not interrupt Explore / Auto-play
- [x] Explore is interrupted by keypresses, mouse clicks, monster in view...
- [x] Auto-play is interrupted by keypresses and mouse clicks
- [x] Recording is interrupted by spacebar, even (especially) in the middle of combat and animations (like throwing darts)
- [x] Mouse and keyboard behave "normally" both during gameplay and recording view. I haven't found a situation when the game didn't react the way I intended.

The comment in #33 "If i spam 's' a few times to search/rest, the action lags behind + doesn't get interrupted while an enemy is near" seems a separate issue, not addressed by this patch. The user wants some keyboard input events _intentionally discarded_ in some situations (e.g monster sighting).